src/main/java/org/testeditor/fixture/web/WebDriverFixture.java

```diff
@@ -660,7 +660,21 @@ public void submit(String elementLocator, LocatorStrategy locatorType) throws Fi
     @FixtureMethod
     public String readValue(String elementLocator, LocatorStrategy locatorType) throws FixtureException {
         WebElement element = getWebElement(elementLocator, locatorType);
-        return element.getText();
+        
+        String readValueThroughAttribute = element.getAttribute("value");
+        String readValueThroughText = element.getText();
+        String value = null;
+        // Selenium getText() vs. getAttribute("value") see https://sqa.stackexchange.com/questions/24463/selenium-webdriver-gettext-vs-getattribute
+        if (!readValueThroughText.isEmpty()) {
+            value = readValueThroughText;
+        } else { 
+            if (!readValueThroughAttribute.isEmpty()) {
+                value = readValueThroughAttribute;
+            } else {
+                throw new FixtureException("No value found for element " + elementLocator);
```

Comment:
please use extended parameter of fixture exception to pass additional info (see last pull request of yours)